operator/internal/manifests/internal/config/loki-config.yaml

```diff
@@ -5,9 +5,44 @@ chunk_store_config:
     enable_fifocache: true
     fifocache:
       max_size_bytes: 500MB
+common:
+  storage:
+    {{ if .AzureObjectStorage }}
```

Comment:
Does it make sense to use `{{ with .AzueObjectStore }}` instead of `if`? While this introduces another `{{ end }}` block for each one it should reduce clutter because you can leave out the `.AzureObjectStore` variable from all the attribute accesses. Similar for all the other config blocks.

Reply:
While `with` does make sense, I don't think I can do an `if else` chain with it (unless I'm reading the docs wrong). The reason I opted for `if else` here is because you can only have one storage configuration. So even though there are multiple stop gaps (including there only being one secret), I had done it this way as extra insurance.

Reply:
That's right, there's no `with else`, that's what I meant with "introducing another `{{ end }}`" and yes the downside of this would be that the template would allow having more than one configuration, which the operator currently treats as invalid.

While this is nice to have, because it should prevent the situation where you have more than one storage configuration, I think it makes sense to bake as little logic into the templating as possible and have this handled in the "proper code" instead. And, as you already pointed out, this is already done. I think having simple-to-read templates outweighs the advantage of having another insurance.